src/sentry/static/sentry/app/views/performance/trends/changedTransactions.tsx

```diff
@@ -0,0 +1,298 @@
+  const queryKey = getSelectedQueryKey(trendChangeType);
+  const offsetString = decodeScalar(location.query[queryKey]);
+  const offset = offsetString ? parseInt(offsetString, 10) : 0;
+  if (!transactions || !transactions.length || offset >= transactions.length) {
+    return undefined;
+  }
+
+  const transaction = transactions[offset].transaction;
+  return transaction;
+}
+
+function handleChangeSelected(
+  location: Location,
+  trendChangeType: TrendChangeType,
+  transactions?: NormalizedTrendsTransaction[]
+) {
+  return function updateSelected(transaction?: NormalizedTrendsTransaction) {
+    const queryKey = getSelectedQueryKey(trendChangeType);
+    const offset = transaction ? transactions?.indexOf(transaction) : -1;
+    const query = {
+      ...location.query,
+    };
+    if (!offset || offset < 0) {
+      delete query[queryKey];
+    } else {
+      query[queryKey] = String(offset);
+    }
+    browserHistory.push({
+      pathname: location.pathname,
+      query,
+    });
+  };
+}
+
+function ChangedTransactions(props: Props) {
+  const {location, trendChangeType, organization} = props;
+  const trendView = props.trendView.clone();
+  modifyTrendView(trendView, location, trendChangeType);
+
+  return (
+    <StyledPanel>
+      <DiscoverQuery
+        eventView={trendView}
+        orgSlug={organization.slug}
+        location={location}
+        trendChangeType={trendChangeType}
+      >
+        {({isLoading, tableData}) => {
+          const eventsTrendsData = (tableData as unknown) as TrendsData;
+          const events = normalizeTrendsTransactions(
+            (eventsTrendsData &&
+              eventsTrendsData.events &&
+              eventsTrendsData.events.data) ||
+              []
+          );
+          const selectedTransaction = getSelectedTransaction(
+            location,
+            trendChangeType,
+            events
+          );
+
+          const results = eventsTrendsData && eventsTrendsData.stats;
+          const transactionsList = events && events.slice ? events.slice(0, 5) : [];
+
+          return (
+            <React.Fragment>
+              <ChartContainer>
+                <Chart
+                  statsData={results}
+                  query={trendView.query}
+                  project={trendView.project}
+                  environment={trendView.environment}
+                  start={trendView.start}
+                  end={trendView.end}
+                  statsPeriod={trendView.statsPeriod}
+                  transaction={selectedTransaction}
+                  isLoading={isLoading}
+                  {...props}
+                />
+              </ChartContainer>
+              <TransactionsList>
+                {transactionsList.map((transaction, index) => (
+                  <TrendsListItem
+                    trendView={trendView}
+                    organization={organization}
+                    transaction={transaction}
+                    key={index}
```

Comment:
Do these items ever resort based on user changes? If so using the index for a key won't work well.

Reply:
I'll take a look 👍